test/test_utils/sagemaker.py

```diff
@@ -226,9 +226,11 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
     sagemaker_regions_list = f"--sagemaker-regions {region_list_str}"
 
     remote_pytest_cmd = (
-        f"pytest -rA {integration_path} --region {region} --processor {processor} {docker_base_arg} "
-        f"{sm_remote_docker_base_name} --tag {tag} {framework_version_arg} {framework_version} "
-        f"{aws_id_arg} {account_id} {instance_type_arg} {instance_type} {efa_flag} {sagemaker_regions_list} --junitxml {test_report}"
+        f"pytest -n=logical --dist worksteal -rA {integration_path} --region {region} "
```

Comment:
Can we add some comments, especially to explain `-n=logical` and `--dist`, to ensure that contributors understand the importance of the commands?

Reply:
Sure, I'll add a comment. I checked to see whether we had comments in other places for pytest args, and decided not to add explanatory comments because that was less work.

However explaining why we're adding certain pytest args for each type of test would inform engineers making any further changes in the future.

Explaining it here as well -
- auto vs logical, and test distribution algorithms - https://pytest-xdist.readthedocs.io/en/stable/distribution.html
- `-n=logical` does nothing on CI/CD, because CodeBuild already provides single-threaded vCPUs at the OS-level to jobs, which means that any parallelization already splits work into one worker per vCPU when using `-n=auto`. However, using `-n=auto` outside CodeBuild (such as on a personal computer or EC2 instance) spawns `num_cpu_cores` parallel worker processes, while using `-n=logical` spawns `num_cpu_cores * 2` worker processes.
- The default `--dist=load` option splits all tests into queues between the available workers, and then lets them handle their own workload. `--dist=worksteal` allows any processes that finish their tests faster to steal tests from the queues for slower workers. This is helpful when our automation needs to run, say, 2 tests while skipping 8 other tests, and we want to have the 2 tests actually be split between workers instead of running the 2 test sequentially because they were scheduled onto the same worker process queue.